### PR TITLE
Changes to allow DB configuration and scaling

### PR DESCRIPTION
**Pull request by mfcrocker**
# Motivation and Context
It turns out that the database parameters in application-cloud.yml are overridden by CloudFoundry. This changes that!

# What has changed
`DataSourceConfiguration` has been added which forces Spring Boot to use the config rather than CloudFoundry's one from buildpacks

# How to test?
You'll need to push to CloudFoundry to see the impact of these changes. The easiest way is to set `spring.datasource.tomcat.max-idle` to a higher number than `max-active`; this will cause Spring Boot to log out a message just after the service starts that max-idle is higher than max-active and give you the figures in the .yml. If it says it's setting it to 4, it's broken (these are the defaults CF was injecting)

### CODE REVIEW

**Comment by mfcrocker** [comment]
[Optimise DB Connection Settings](https://trello.com/c/SwXkyTRo/44-optimise-db-connection-settings)

File: src/main/java/uk/gov/ons/ctp/response/sample/SampleSvcApplication.java [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package uk.gov.ons.ctp.response.sample;
 
 import net.sourceforge.cobertura.CoverageIgnore;
+
+import org.apache.tomcat.jdbc.pool.DataSource;
```
Comment (tej99) [inline src/main/java/uk/gov/ons/ctp/response/sample/SampleSvcApplication.java]: Unused import

File: .checkstyle [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (tej99) [inline .checkstyle]: What's this?

File: src/main/java/uk/gov/ons/ctp/response/sample/SampleSvcApplication.java [hunk 3]
```diff
@@ -1,6 +1,7 @@
 package uk.gov.ons.ctp.response.sample;
 
 import net.sourceforge.cobertura.CoverageIgnore;
+
```
Comment (VibhorMehra) [inline src/main/java/uk/gov/ons/ctp/response/sample/SampleSvcApplication.java]: Not here but around line 57, we're missing an auto wired DataSource

**Pull request merged**